cmake/pydarknet_utils.cmake

```diff
@@ -0,0 +1,45 @@
+macro(GET_OS_INFO)
+    string(REGEX MATCH "Linux" OS_IS_LINUX ${CMAKE_SYSTEM_NAME})
+    string(REGEX MATCH "Darwin" OS_IS_MACOS ${CMAKE_SYSTEM_NAME})
+    set(PYDARKNET_LIB_INSTALL_DIR "lib${LIB_SUFFIX}")
+    set(PYDARKNET_INCLUDE_INSTALL_DIR
+        "include/${PROJECT_NAME_LOWER}-${PYDARKNET_VERSION_MAJOR}.${PYDARKNET_VERSION_MINOR}")
+endmacro(GET_OS_INFO)
```

Comment:
Nothing wrong with this macro, but one of the things I came across while digging into the cmake documentation was around predefined os variables (e.g. `APPLE`).

The other standard variables include `PROJECT_VERSION*`, which is set via `project(...)`. This then enables one to standardize and cookie cutter the cmake code without needing to hardcode the project name in lots of places. Though `<project-name>_VERSION*` is also be available if wanted. See also https://cmake.org/cmake/help/v3.18/variable/PROJECT-NAME_VERSION.html?highlight=project

I'm just mentioning these things for future reference.

Reply:
Perfect!  Our CMakeFiles have been around for years and it's nice that they have added a bunch of default directives for us to just use.  We can modernize our templates as time goes on.  One of the things I am most excited about is our templating is pretty uniform outside of the specific libraries and CUDA support are needed for each repo.

Reply:
If you want to start applying this fix to all of the standardized CMake templates in our updated repos, I won't object